improver/spotdata/spot_extraction.py

```diff
@@ -92,7 +92,11 @@ def extract_coordinates(self, neighbour_cube):
         index_constraint = iris.Constraint(grid_attributes_key=["x_index", "y_index"])
         coordinate_cube = neighbour_cube.extract(method & index_constraint)
         if coordinate_cube:
-            coordinate_cube.data = np.rint(coordinate_cube.data).astype(int)
+            data = coordinate_cube.data
+            data = np.rint(data).astype(int)
+            # sort the spot sites for better memory accessing later
+            ind = np.lexsort((data[1], data[0]))
+            coordinate_cube.data = data[:, ind]
```

Comment:
This should go into `neighbour_finding.py` rather than here to avoid sorting on each extraction.

Reply:
hmm good point, should I move the rint and casting there too?

Reply:
Just having another look - I don't think the casting can go here because the vertical displacement isn't an int, but I think the rint on the x and y dimensions can happen here

Reply:
Sorry, I marked to many lines. I meant the `lexsort` part.